modules/developer_manual/pages/testing/unit-testing.adoc

```diff
@@ -78,21 +79,96 @@ You can find more information in {phpunit-docs-url}[the PHPUnit documentation].
 
-* Change into `webroot` and run `make help` to see tests and parameters
-available.
+You should see output similar to the below example.
 
-Testing apps
+[source,console]
+----
+make test-acceptance-api	run API acceptance tests
+make test-acceptance-cli	run CLI acceptance tests
+make test-acceptance-webui	run webUI acceptance tests
+make test-js-debug		run Javascript tests in debug mode (continuous)
+make test-js			run Javascript tests
+make test-php-phan		run PHP phan static code analyzer
+make test-php-phpstan		run PHP phpstan static code analyzer
+make test-php-style-fix		run PHP code style checks and fix any issues found
+make test-php-style		run PHP code style checks
+make test-php-unit		run all PHP tests
+make test			run all tests
+----
+
+==== Testing Apps
 
-* To run the tests for a specific app with the provided PHPUnit version,
-change into `<webroot>/apps/<appname>` and
+To run the tests for a specific app with the provided PHPUnit version: 
 
+. Change into the apps directory, in the root directory of your ownCloud installation
+. Then change into the directory of the app that you want to test. 
+. Run the follow command. 
++
+--
+[source,console]
 ----
 make test-php-unit
 ----
 
-[[writing-php-unit-tests]]
+Here's an example of running the command in {encryption-app-url}[the encryption app]:
+
+[source,console]
+----
+php -d zend.enable_gc=0 ../../lib/composer/bin/phpunit --configuration ./phpunit.xml --testsuite unit
+PHPUnit 6.5.14 by Sebastian Bergmann and contributors.
+
+Runtime:       PHP 7.2.19-0ubuntu0.19.04.2
+Configuration: /var/www/owncloud-community/apps/encryption/phpunit.xml
+Error:         No code coverage driver is available
+
+a test string for helloWelcome to your ownCloud account!
+
+This is just an example file for developers and git users. 
+The packaged and released versions will come with better examples.
+
+a test string for world.a foo testLets modify again3Welcome to your ownCloud account!
+
+This is just an example file for developers and git users. 
+The packaged and released versions will come with better examples.
+
+a test string for world.a foo testLets modify again3Welcome to your ownCloud account!
+
+This is just an example file for developers and git users. 
+The packaged and released versions will come with better examples.
+
+a test string for world.............................................................  63 / 251 ( 25%)
+............................................................... 126 / 251 ( 50%)
+............................................................... 189 / 251 ( 75%)
+..............................................................  251 / 251 (100%)
+
+Time: 29.21 seconds, Memory: 32.00MB
+
+OK (251 tests, 724 assertions)
+----
+--
+
+
+[NOTE] 
+====
+Apps that are part of core *do not* have their own Makefile; these apps are: 
```

Comment:
How are we going to maintain this list when it changes?
Is there somewhere that the list of "apps that are actually part of core" is already documented, and we can link  to it?

Reply:
I thought that there was some element of contention about what was and what was not a core app over the last 6 months. However, after a bit of grokking, I found https://github.com/owncloud/docs/pull/1411/files#diff-c6969b2f582ea066559074e8cf3b42e0R61, which says:

> Third party apps are all apps that are not distributed by [ownCloud](https://marketplace.owncloud.com/publishers/owncloud).

That list is currently:

- activity
- admin_audit
- announcementcenter
- calendar
- contacts
- customgroups
- diagnostics
- enterprise_key
- files_antivirus
- files_classifier
- files_external_dropbox
- files_external_ftp
- files_ldap_home
- files_mediaviewer
- files_pdfviewer
- files_primary_s3
- files_texteditor
- firewall
- gallery
- guests
- impersonate
- market
- oauth2
- objectstore
- password_policy
- phoenix
- ransomware_protection
- sharepoint
- systemtags_management
- templateeditor
- twofactor_totp
- user_ldap
- user_shibboleth
- windows_network_drive
- workflow